courseScheduleII.js

```diff
@@ -0,0 +1,110 @@
+/*
+For this approach, we can represent the input as a graph where in order to take a course, 
+you have to complete a prereq, which is the same as two nodes conntected by an edge.
+ 
+In order to traverse to the next node, you have to start with the one connected to it prior. 
+As you can see, it's going to be a directed graph, since we can also travel in one direction.
+
+Since we are trying to find a valid ordering for all these courses that respects our prereqs, 
```

Comment:
You really didn't explain anything about HOW this algorithm works. Conceptually, what is it doing?

If you don't know the answer, I suggest that you find a YouTube video and/or work through this code on an example input. That will help you develop a deeper understanding of what this algorithm does. Then, try again at explaining what it's doing.

Here are 4 links I found from just a quick search (I haven't watched these so I don't know if they're good):
https://www.youtube.com/watch?v=rZv_jHZva34
https://www.youtube.com/watch?v=73sneFXuTEg
https://www.youtube.com/watch?v=iI4HhxTZWO8
https://www.youtube.com/watch?v=XMiIk1B0v-w